### PR TITLE
Doc 6412a

### PR DESCRIPTION
**Pull request by tonyjhillman**

### CODE REVIEW

File: modules/fts/pages/fts-geospatial-queries.adoc [hunk 1]
```diff
@@ -6,21 +6,32 @@ Geospatial queries return documents that contain location-data.
...
 The location determines the center of a circle whose radius-length is the specified distance.
 Documents are returned if they reference a location within the circle.
-* Two longitude-latitude coordinate pairs.
-These are respectively taken to indicate the top left and bottom right corners of a bounding box.
+
+* Two latitude-longitude coordinate pairs.
+These are respectively taken to indicate the top left and bottom right corners of a _rectangular_ bounding box.
+Documents are returned if they reference a location within the box.
+
+* An array of three of more latitude-longitude coordinate pairs.
```
Comment (sreekanth-cb) [inline modules/fts/pages/fts-geospatial-queries.adoc]: three of more => three or more  ?

File: modules/fts/pages/fts-geospatial-queries.adoc [hunk 2]
```diff
@@ -6,21 +6,32 @@ Geospatial queries return documents that contain location-data.
...
+Areas and locations are represented by means of _latitude_-_longitude_ coordinate pairs.
 
-This allows an application, based on the user's input of a particular coordinate, address, or property name, to derive a list of _points of interest_ that lie within a specified distance of the user-referenced location; and display these for the user's benefit.
+This allows an application, based on the user's input of a particular coordinate, address, or property name, to derive a list of _points of interest_ that lie within an area that surrounds the user-referenced location; and display these for the user's benefit.
 
-The location-data provided by a geospatial query can be either of the following:
+A geospatial query can specify an area by means of:
 
-* A location, specified as a longitude-latitude coordinate pair; and a distance, in miles.
+* A location, expressed as a latitude-longitude coordinate pair; and a distance, in miles.
```
Comment (sreekanth-cb) [inline modules/fts/pages/fts-geospatial-queries.adoc]: Formats of distance units supported:
"5in" "5inch" "7yd" "7yards" "9ft" "9feet" "11km" "11kilometers" "3nm" "3nauticalmiles" "13mm" "13millimeters" "15cm" "15centimeters" "17mi" "17miles" "19m" "19meters"

If the unit cannot be determined, the entire string is parsed and the unit of meters is assumed.

File: modules/fts/pages/fts-geospatial-queries.adoc [hunk 3]
```diff
@@ -6,21 +6,32 @@ Geospatial queries return documents that contain location-data.
...
 Documents are returned if they reference a location within the circle.
-* Two longitude-latitude coordinate pairs.
-These are respectively taken to indicate the top left and bottom right corners of a bounding box.
+
+* Two latitude-longitude coordinate pairs.
+These are respectively taken to indicate the top left and bottom right corners of a _rectangular_ bounding box.
+Documents are returned if they reference a location within the box.
+
+* An array of three of more latitude-longitude coordinate pairs.
+Each of the pairs is taken to indicate one corner of a _polygonal_ bounding box.
```
Comment (sreekanth-cb) [inline modules/fts/pages/fts-geospatial-queries.adoc]: _polygonal_ bounding box looks wrong,
_polygonal_ bounding area sounds better?

Similarly,
a location within the polygon bounding area

File: modules/fts/pages/fts-geospatial-queries.adoc [hunk 4]
```diff
@@ -6,21 +6,32 @@ Geospatial queries return documents that contain location-data.
...
 Documents are returned if they reference a location within the box.
 
-To be successful, a geospatial query must reference an index within which the _geopoint_ type mapping has been applied to the field containing the target longitude-latitude coordinate pair.
+To be successful, a geospatial query must reference an index within which the _geopoint_ type mapping has been applied to the field containing the target latitude-longitude coordinate pair.
+
+Geospatial queries return _all_ documents whose locations are within the query-specified area.
+To specify _holes_ within the area, so that one or more subsets of returned documents can be omitted from the final results, boolean queries should be applied to the set of documents returned by the geospatial query.
+See xref:fts:fts-query-types.adoc[Query Types].
+
+For any geospatial query, latitude-longitude coordinate pairs can also be specified as _geohashes_, as demonstrated in xref:fts:fts-geospatial-queries.adoc#specifying-polygon-points[Specifying Polygon Points], below.
```
Comment (sreekanth-cb) [inline modules/fts/pages/fts-geospatial-queries.adoc]: latitude-longitude can be expressed in any of the 4 accepted formats and this is applicable not only to polygon query but all geo queries.
ref - https://blog.couchbase.com/introducing-geo-bounded-polygon-query-in-full-text-search/
You may find all 4 format types there..
This needs to be documented somewhere.

File: modules/fts/pages/fts-geospatial-queries.adoc [hunk 5]
```diff
@@ -226,6 +237,127 @@ A subset of formatted output might appear as follows:
           .
 ----
 
+[#creating_geospatial_rest_query_polygon_based]
+== Creating a Query: Polygon-Based
+
+The following query body uses an array, containing multiple latitude-longitude pairs, to specify each of the corners of a _polygonal_ bounding box &#8212; known as _polygon points_.
```
Comment (sreekanth-cb) [inline modules/fts/pages/fts-geospatial-queries.adoc]: bounding box -> bounding area/polygon?

File: modules/fts/pages/fts-query-types.adoc [hunk 6]
```diff
@@ -379,13 +379,17 @@ A phrase query is also demonstrated by means of the Java SDK, in xref:2.7@java-s
 == Geospatial Queries
 
 _Geospatial_ queries return documents that each specify a geographical location.
-Each query contains either:
+The location-data provided by a geospatial query can be any of the following:
 
-* A single _longitude_-_latitude_ coordinate pair; and a _distance_ value, in miles, which determines a radius measured from the location specified by the coordinate pair.
-Documents are returned if they specify (by means of a longitude-latitude coordinate pair) a location that lies within the radius.
+* A location, specified as a longitude-latitude coordinate pair; and a distance, in miles.
```
Comment (sreekanth-cb) [inline modules/fts/pages/fts-query-types.adoc:384]: miles -> same feedback...on distance unit.

File: modules/fts/pages/fts-query-types.adoc [hunk 7]
```diff
@@ -379,13 +379,17 @@ A phrase query is also demonstrated by means of the Java SDK, in xref:2.7@java-s
...
+The location determines the center of a circle whose radius-length is the specified distance.
+Documents are returned if they reference a location within the circle.
 * Two longitude-latitude coordinate pairs.
-These are respectively taken to indicate the upper left and lower right corners of a bounding box.
-Documents are returned if they specify a location that lies within the bounding box.
+These are respectively taken to indicate the top left and bottom right corners of a _rectangular_ bounding box.
+Documents are returned if they reference a location within the box.
+* An array of three of more longitude-latitude coordinate pairs.
+Each of the pairs is taken to indicate one corner of a _polygonal_ bounding box.
```
Comment (sreekanth-cb) [inline modules/fts/pages/fts-query-types.adoc:391]: same bounding box comment..

**Review by sreekanth-cb — CHANGES_REQUESTED** [review]
left some comments, please let me know of any clarifications here,

File: modules/fts/pages/fts-geospatial-queries.adoc (hunk 1 above)
Comment (tonyjhillman) [reply modules/fts/pages/fts-geospatial-queries.adoc]: Thanks, done.

File: modules/fts/pages/fts-geospatial-queries.adoc (hunk 3 above)
Comment (tonyjhillman) [reply modules/fts/pages/fts-geospatial-queries.adoc]: I've just tried using 'circle', 'rectangle', and 'polygon', for simplicity. I've also cleaned up my references to areas and locations, so hopefully it looks better now.

File: modules/fts/pages/fts-geospatial-queries.adoc (hunk 4 above)
Comment (tonyjhillman) [reply modules/fts/pages/fts-geospatial-queries.adoc]: I've put in a new section, to cover this.

File: modules/fts/pages/fts-query-types.adoc (hunk 7 above)
Comment (tonyjhillman) [reply modules/fts/pages/fts-query-types.adoc:391]: Thanks, done as per comments above.

File: modules/fts/pages/fts-query-types.adoc (hunk 6 above)
Comment (tonyjhillman) [reply modules/fts/pages/fts-query-types.adoc:384]: Thanks, done.

File: modules/fts/pages/fts-geospatial-queries.adoc (hunk 5 above)
Comment (tonyjhillman) [reply modules/fts/pages/fts-geospatial-queries.adoc]: Used polygon, here.

**Comment by tonyjhillman** [comment]
I've moved things around a bit, and addressed the text throughout, to clarify if possible. Let me know what you think. Thanks!

File: modules/fts/pages/fts-query-types.adoc (hunk 7 above)
Comment (sreekanth-cb) [reply modules/fts/pages/fts-query-types.adoc:391]: feels like the "_polygonal_ bounding box" is a bit confusing to others.. :-) 
what about _polygonal_ bounding area?  Its used in one more place.

Comment (sreekanth-cb) [reply modules/fts/pages/fts-query-types.adoc:391]: _polygonal_ bounding box. => polygon here too?

**Review by sreekanth-cb — CHANGES_REQUESTED** [review]

File: modules/fts/pages/fts-geospatial-queries.adoc [hunk 8]
```diff
@@ -53,11 +63,167 @@ Moreover, any other child-field, such as `alt`, is ignored.
...
+
+If the unit cannot be determined, the entire string is parsed, and the distance is assumed to be in _meters_.
+
 [#creating_a_geospatial_index]
 == Creating a Geospatial Index
 
-To be successful, a geospatial query must reference an index that applies the _geopoint_ type mapping to the field containing the longitude-latitude coordinate pair.
-This can be achieved by means of the Couchbase Web Console.
+To be successful, a geospatial query must reference an index that applies the _geopoint_ type mapping to the field containing the latitude-longitude coordinate pair.
+This can be achieved by means of Couchbase Web Console.
```
Comment (sreekanth-cb) [inline modules/fts/pages/fts-geospatial-queries.adoc:226]: or directly through the REST end points for index creation/updation.

**Review by sreekanth-cb — COMMENTED** [review]
left some minor comments.. Otherwise looks good to me..

**Pull request merged**